runtime/doc/lua.txt

```diff
@@ -2366,7 +2366,26 @@ vim.trim({s})                                                     *vim.trim()*
       • https://www.lua.org/pil/20.2.html
 
 vim.validate({opt})                                           *vim.validate()*
-    Validates a parameter specification (types and values). Specs are
+    Validate function arguments.
+
+    This function has two valid forms:
+    1. vim.validate(name: str, value: any, type: string, optional?: bool)
+    2. vim.validate(spec: table)
+
+    Form 1 validates that argument {name} with value {value} has the type
+    {type}. {type} must be a value returned by |lua-type()|. If {optional} is
+    true, then {value} may be null. This form should be preferred for simple
+    cases.
```

Comment:
should we hint that form 1 is faster?
```suggestion
    true, then {value} may be nil. Performance is comparable to assert().
```

Reply:
feec3da5dcc742ba42a552c3549ccaf02f569ba8